src/devices/vdml_ext_adi.c

```diff
@@ -408,3 +411,32 @@ int32_t ext_adi_gyro_shutdown(ext_adi_gyro_t gyro) {
 	vexDeviceAdiPortConfigSet(device->device_info, adi_port, E_ADI_TYPE_UNDEFINED);
 	return_port(smart_port, 1);
 }
+
+ext_adi_potentiometer_t ext_adi_potentiometer_init(uint8_t smart_port, uint8_t adi_port, adi_potentiometer_type_e_t potentiometer_type) {
+	transform_adi_port(adi_port);
+	claim_port_i(smart_port, E_DEVICE_ADI);
+
+	adi_data_s_t* const adi_data = &((adi_data_s_t*)(device->pad))[adi_port];
+	adi_data->potentiometer_data.potentiometer_type = potentiometer_type;
+
+	return_port(smart_port - 1, merge_adi_ports(smart_port - 1, adi_port + 1));
+}
+
+double ext_adi_potentiometer_get(ext_adi_potentiometer_t potentiometer) {
```

Comment:
```suggestion
double ext_adi_potentiometer_get_angle(ext_adi_potentiometer_t potentiometer) {
```

(do the same with all relevant areas), c++ call should be .get_angle().